iocore/net/UnixNetAccept.cc

```diff
@@ -201,37 +201,54 @@ NetAccept::init_accept(EThread *t)
   t->schedule_every(this, period);
 }
 
-void
-NetAccept::init_accept_per_thread()
+int
+NetAccept::accept_per_thread(int event, void *ep)
 {
-  int i, n;
+  int listen_per_thread = 0;
+  REC_ReadConfigInteger(listen_per_thread, "proxy.config.exec_thread.listen");
 
-  ink_assert(opt.etype >= 0);
-
-  if (do_listen(NON_BLOCKING)) {
-    return;
+  if (listen_per_thread == 1) {
+    if (do_listen(NON_BLOCKING)) {
```

Comment:
I think I am missing something with do_listen().  It looks like it returns the > 0 on success.  Why are you returning from both accept_per_thread() and init_accept_per_thread() in each option of listen_per_thread when it is successful?